### PR TITLE
#53 - forgot password page frontend

### PR DESCRIPTION
**Pull request by JokeUrSelf**
Added UI to this simple page

### CODE REVIEW

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Missing screenshot in the pr description
2. I don't like this input stretched on whole page. Why we cant just make some tile on the center of the screen with one field?


image

3. I can click all the time the button "Zmień hasło" - maybe we should block it somehow? Or I don't know

4. Reading the title of pr, I expected that also this page will have UI

image

**Comment by JokeUrSelf** [comment]
> 1. Missing screenshot in the pr description
> 2. I don't like this input stretched on whole page. Why we cant just make some tile on the center of the screen with one field?
> 
> image
> 
> 3. I can click all the time the button "Zmień hasło" - maybe we should block it somehow? Or I don't know
> 4. Reading the title of pr, I expected that also this page will have UI
> 
> image

1. Will do
2. Already fixed
3. To you want to block while the request is being processed, or for some range of time, like 60 seconds? The timer option will require changes to backend, feels complicated.
4. Makes sense, will add.

**Review by AleksandraKozubal — COMMENTED** [review]
Input is still stretched

File: resources/js/Pages/Auth/ForgotPassword.vue [hunk 1]
```diff
@@ -1,30 +1,43 @@
...
 
 <template>
-  <form @submit.prevent="submit">
-    <div>
-      <label for="email">Email:</label>
-      <input v-model="form.email" name="email" type="email">
-      <button type="submit">Reset?</button>
-      <div v-if="errors.email">{{ errors.email }}</div>
+  <Head>
+    <title>Profile</title>
```
Comment (AleksandraKozubal) [inline resources/js/Pages/Auth/ForgotPassword.vue:26]:
```suggestion
    <title>Zmiana hasła</title>
```

File: resources/js/Pages/Auth/ForgotPassword.vue [hunk 2]
```diff
@@ -1,30 +1,43 @@
...
+    <meta name="Zmień hasło" content="User profile">
```
Comment (AleksandraKozubal) [inline resources/js/Pages/Auth/ForgotPassword.vue:27]: Why is it half in eng and half in pl?

**Review by AleksandraKozubal — COMMENTED** [review]
Please change title to "Zmiana hasła" and meta tags to be appropriate

**Comment by EwelinaSkrzypacz** [comment]
Conflicts :(

**Review by EwelinaSkrzypacz — COMMENTED** [review]
For me, we should have some link to back on the main page

image

File: resources/js/components/Home/BackgroundEffect.vue [hunk 3]
```diff
@@ -10,7 +10,7 @@ onUnmounted(() => window.removeEventListener('scroll', setBannerVisibility))
 </script>
 
 <template>
-  <div :style="{marginTop: `${margin}rem`}" class="fixed blur-2xl size-7/12 -z-10">
+  <div :style="{marginTop: `${margin}rem`}" class="fixed blur-2xl size-7/12 -z-50">
```
Comment (kamilpiech97) [inline resources/js/components/Home/BackgroundEffect.vue:13]: It is for what?

File: resources/js/components/Common/Banner.vue [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<script setup lang="ts">
+import { XMarkIcon } from '@heroicons/vue/20/solid'
+defineProps<{text:string}>()
+defineEmits<{ click: [] }>()
```
Comment (kamilpiech97) [inline resources/js/components/Common/Banner.vue:4]: It is for what?

File: resources/js/components/Home/BackgroundEffect.vue (hunk 3 above)
Comment (JokeUrSelf) [reply resources/js/components/Home/BackgroundEffect.vue:13]: I have no idea

File: resources/js/components/Common/Banner.vue (hunk 4 above)
Comment (JokeUrSelf) [reply resources/js/components/Common/Banner.vue:4]: To import X icon and make X button functional

**Comment by JokeUrSelf** [comment]
> For me, we should have some link to back on the main page
> 
> image

That's gonna be in the header, Dominik works on that

File: resources/js/components/Common/Banner.vue (hunk 4 above)
Comment (kamilpiech97) [reply resources/js/components/Common/Banner.vue:4]: I mean `defineEmits`

File: resources/js/components/Home/BackgroundEffect.vue (hunk 3 above)
Comment (kamilpiech97) [reply resources/js/components/Home/BackgroundEffect.vue]: I mean `${margin}`

File: resources/js/components/Common/Banner.vue (hunk 4 above)
Comment (JokeUrSelf) [reply resources/js/components/Common/Banner.vue:4]: I use it to make a function call from banner in a parent

File: resources/js/Pages/Auth/ForgotPassword.vue [hunk 5]
```diff
@@ -1,31 +1,51 @@
 <script setup lang="ts">
+import { useForm } from '@inertiajs/vue3'
+import CustomInput from '@/components/Common/CustomInput.vue'
+import Header from '@/components/Common/Header.vue'
+import Banner from '@/components/Common/Banner.vue'
+import Footer from '@/components/Common/Footer.vue'
+import BackgroundEffect2 from '@/components/Common/BackgroundEffect2.vue'
+import { Head } from '@inertiajs/vue3'
+import { ref } from 'vue'
```
Comment (kamilpiech97) [inline resources/js/Pages/Auth/ForgotPassword.vue:10]:
```suggestion
```

File: resources/js/Pages/Auth/ResetPassword.vue [hunk 6]
```diff
@@ -1,45 +1,47 @@
 <script setup lang="ts">
+import { useForm } from '@inertiajs/vue3'
+import { Head } from '@inertiajs/vue3'
+import Header from '@/components/Common/Header.vue'
+import Footer from '@/components/Common/Footer.vue'
+import CustomInput from '@/components/Common/CustomInput.vue'
+import BackgroundEffect2 from '@/components/Common/BackgroundEffect2.vue'
```
Comment (kamilpiech97) [inline resources/js/Pages/Auth/ResetPassword.vue:8]:
```suggestion
```

File: resources/js/Pages/Home.vue [hunk 7]
```diff
@@ -29,3 +29,9 @@ provide('isLoginRef', isLogin)
     <Footer />
   </div>
 </template>
+
+<style scoped>
+html {
+  overflow-y: scroll;
+}
+</style>
```
Comment (kamilpiech97) [inline resources/js/Pages/Home.vue:37]: Why here? Not in layout?

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: resources/js/components/Home/BackgroundEffect.vue (hunk 3 above)
Comment (JokeUrSelf) [reply resources/js/components/Home/BackgroundEffect.vue]: It changes depending on scroll

File: resources/js/Pages/Home.vue (hunk 7 above)
Comment (JokeUrSelf) [reply resources/js/Pages/Home.vue:37]: This is the only page where it is going to be used, to fix an edge case where page resizes itself when search is opened

**Review by EwelinaSkrzypacz — APPROVED** [review]
I'm not a big fan of this background, but we can change it later

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**